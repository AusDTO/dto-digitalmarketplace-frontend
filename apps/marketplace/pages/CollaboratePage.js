import React from 'react'
import { withRouter, Switch, Route, Redirect } from 'react-router-dom'
import PropTypes from 'prop-types'
import CollaborateView from '../components/Collaborate/Landing'
import { rootPath } from '../routes'

export const CollaboratePageContainer = props => {
  const { match } = props

  return (
    <div id="Collaborate-page">
      <Switch>
        <Route exact path={match.url} render={() => <CollaborateView {...props} />} />
        <Redirect to={{ pathname: `${rootPath}/collaborate` }} />
      </Switch>
    </div>
  )
}

CollaboratePageContainer.propTypes = {
  match: PropTypes.object
}

CollaboratePageContainer.defaultProps = {
  match: () => ({ url: `${rootPath}/collaborate` })
}

export default withRouter(CollaboratePageContainer)
