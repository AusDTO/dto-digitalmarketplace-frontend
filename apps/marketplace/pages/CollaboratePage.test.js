import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import { Provider } from 'react-redux'
import expect from 'expect'
import configureStore from '../store'
import { CollaboratePageContainer } from './CollaboratePage'

jest.mock('shared/Icon/_getIcons')

describe('Test suite for CollaboratePageContainer component', () => {
  const state = {}

  const store = configureStore()
  test('CollaboratePageContainer renders without errors', () => {
    const renderer = new ReactShallowRenderer()
    const tree = renderer.render(
      <Provider store={store}>
        <CollaboratePageContainer {...state} />
      </Provider>
    )

    expect(tree).toMatchSnapshot()
  })
})
