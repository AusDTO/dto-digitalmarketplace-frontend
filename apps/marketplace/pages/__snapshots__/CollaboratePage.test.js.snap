// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for CollaboratePageContainer component CollaboratePageContainer renders without errors 1`] = `
<CollaboratePageContainer
  match={[Function]}
/>
`;
